### ISSUE TITLE
1.9.3 Nolonger connects using keys

### ISSUE THREAD

**Issue by diversemix** [connectbot/connectbot#656]
I can no longer connect using the stored keys. I've tried recreating and using different key lengths but to no avail

**Comment by kruton** [comment]
Sorry, bug with pre-API 19 versions. Rolling out a fix now; follow #654

**Comment by alescdb** [comment]
The last version that connect with my key only servers is v1.9.2 (git-v1.9.2 19 Feb 2017) every update since made the same errors (same with latest git-v1.9.4 from minutes ago) :

![screenshot_20181104-141704](https://user-images.githubusercontent.com/5646825/47964767-a2e04100-e03e-11e8-8aca-6489f1e0cbc6.jpg)

On the servers side noticeable settings are : 

```
PasswordAuthentication no
ChallengeResponseAuthentication no
```

**Comment by kruton** [comment]
Interesting. What type of key are you using?

**Comment by alescdb** [comment]
RSA 1024 SHA256, I'll try to generate a new one and let you know.

**Comment by alescdb** [comment]
Juste tried with a brand new 4096 RSA key (generated with `ssh-keygen -b 4096`), same error (works with 1.9.2, not with higher version).

**Comment by hagau** [comment]
This might be related to https://github.com/openssh/openssh-portable/commit/4ba0d54794814ec0de1ec87987d0c3b89379b436  
See https://github.com/open-keychain/sshlib/commit/686e8d44d80ea65167972f88a0a9c440559e5fd4 for a fix

**Comment by alescdb** [comment]
If it can helps here is a screenshot of a successfull connection with v1.9.2 (in french, sorry) : 

![image](https://user-images.githubusercontent.com/5646825/47991671-20fa2180-e0eb-11e8-94f4-105030ecaee4.png)

Logs on servers are not very useful :

- v1.9.2, `Accepted publickey` then `session opened`, 
- v1.9.4, nothing when I connect and `Received disconnect from` then `Disconnected from authenticating user`

**Comment by kruton** [comment]
Ah, looks like it only fails with OpenSSH 7.8 and newer. The integration tests were running OpenSSH 7.7.

**Comment by kruton** [comment]
Should be fixed by 18e8fe030eb45d7606adff22292e2b0b6cc5d1d4.

**Comment by alescdb** [comment]
> Should be fixed by [18e8fe0](https://github.com/connectbot/connectbot/commit/18e8fe030eb45d7606adff22292e2b0b6cc5d1d4).

I do confirm 1.9.5 works (OpenSSH 7.9)
